pmd-java/src/main/java/net/sourceforge/pmd/lang/java/typeresolution/MethodTypeResolution.java

```diff
@@ -510,4 +510,12 @@ public static boolean isSubtypeable(JavaTypeDefinition parameter, JavaTypeDefini
 
         return false;
     }
+
+    public static JavaTypeDefinition boxPrimitive(JavaTypeDefinition def) {
+        if (!def.isPrimitive()) {
+            return null;
+        }
+
+        return JavaTypeDefinition.forClass(BOXED_PRIMITIVE_SUBTYPE_ORDER.get(PRIMITIVE_SUBTYPE_ORDER.indexOf(def.getType())));
```

Comment:
I'm thinking we should change these 2 lists to a single `TreeMap` with primitive classes as keys, and boxed ones as values. This would simplify all of our current usages except maybe for `isSubtypeable` that is the only one that still cares about order,  but can still be used since insertion order is kept.